File: 08 String/Valid Anagram/Anagram.java
import java.util.HashMap;
import java.util.Scanner;

class Solution {
    boolean isAnagram(String s, String t) {
        if (s.length() != t.length())
            return false;
        HashMap<Character, Integer> counter1 = new HashMap<>();
        HashMap<Character, Integer> counter2 = new HashMap<>();
        for (char c : s.toCharArray()) {
            counter1.put(c, counter1.getOrDefault(c, 0) + 1);
        }
        for (char c : t.toCharArray()) {
            counter2.put(c, counter2.getOrDefault(c, 0) + 1);
        }
        return counter1.equals(counter2);
    }
}

public class Anagram {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int tc = sc.nextInt();
        sc.nextLine();
        Solution obj = new Solution();

        while (tc > 0) {
            String s = sc.nextLine().replaceAll("\\s", "");
            String t = sc.nextLine().replaceAll("\\s", "");
            System.out.println(obj.isAnagram(s, t));
            tc--;
        }
        sc.close();
    }
}

File: 08 String/Valid Anagram/Anagram.scala
import scala.collection.mutable.HashMap
import scala.io.StdIn

object Solution {
  def isAnagram(s: String, t: String): Boolean = {
    if (s.length != t.length) {
      false
    } else {
      val counter1 = new HashMap[Char, Int]()
      val counter2 = new HashMap[Char, Int]()
      s.foreach(c => counter1.put(c, counter1.getOrElse(c, 0) + 1))
      t.foreach(c => counter2.put(c, counter2.getOrElse(c, 0) + 1))
      counter1 == counter2
    }
  }
}

object Anagram extends App {
  val tc = StdIn.readInt()
  for (_ <- 1 to tc) {
    val s = StdIn.readLine()
    val t = StdIn.readLine()
    println(Solution.isAnagram(s, t))
  }
}
